### PR TITLE
Update with api

### PR DESCRIPTION
**Pull request by AlexisFLT**
add an horror filter

### CODE REVIEW

**Review by Fbouchezpro — DISMISSED** [review]

File: frontend/src/App.jsx [hunk 1]
```diff
@@ -1,12 +1,32 @@
...
+import React, { useEffect } from "react";
+
 import "./App.css";
 
 function App() {
+  const [movies, setMovies] = React.useState([]);
+  const getMovies = () => {
+    axios
+      .get(
+        "https://api.themoviedb.org/3/movie/popular?api_key=f352c00b6a6dc4de283669f3f964df1a&language=en-US&page=1"
```
Comment (LoicBrassart) [inline frontend/src/App.jsx]: Yout ApiKey should be in a .dotenv file :grimacing:

File: frontend/src/App.jsx [hunk 2]
```diff
@@ -1,12 +1,32 @@
...
+      )
+      .then((response) => response.data)
+      .then((data) => {
+        setMovies(data.results);
+      });
```
Comment (LoicBrassart) [inline frontend/src/App.jsx]: Another, shorter syntax:
```
    axios
      .get("blablablaurlblabla")
      .then(({data}) => {
        setMovies(data.results.filter((movie) => {
          return !movie.genre_ids.includes(27);
        }));
      });
```

File: frontend/src/App.jsx [hunk 3]
```diff
@@ -1,12 +1,32 @@
...
+  };
+  useEffect(() => {
+    getMovies();
+  }, []);
+
+  const filter18 = movies.filter((movie) => {
+    return !movie.genre_ids.includes(27);
+  });
```
Comment (LoicBrassart) [inline frontend/src/App.jsx]: Doing code here is not a good idea, it should be in the useEffect 
See previous comment

**Review by LoicBrassart — CHANGES_REQUESTED** [review]
Not too bad, but a few things to fix ;-)

File: frontend/src/App.jsx (hunk 1 above)
Comment (Fbouchezpro) [reply frontend/src/App.jsx]: Done

File: frontend/src/App.jsx (hunk 2 above)
Comment (Fbouchezpro) [reply frontend/src/App.jsx]: Thx changes done

File: frontend/src/App.jsx (hunk 3 above)
Comment (Fbouchezpro) [reply frontend/src/App.jsx]: See previous :-D

File: frontend/.env.sample [hunk 4]
```diff
@@ -1,2 +1,4 @@
 # call it in React with import.meta.env.VITE_BACKEND_URL
-VITE_BACKEND_URL=http://localhost:5000
\ No newline at end of file
+VITE_BACKEND_URL=http://localhost:5000
+
+SECRET_API_KEY = YOUR_API_KEY
```
Comment (LoicBrassart) [inline frontend/.env.sample]: Nope, it'll have to begin with "VITE_"

File: frontend/package.json [hunk 5]
```diff
@@ -10,6 +10,8 @@
     "fix": "prettier --write . && eslint --fix --ext .jsx ."
   },
   "dependencies": {
+    "axios": "^1.1.3",
+    "dotenv": "^16.0.3",
```
Comment (LoicBrassart) [inline frontend/package.json]: You can remove dotenv from the frontend

**Review by LoicBrassart — CHANGES_REQUESTED** [review]
Small errors to fix

**Review by Fbouchezpro — DISMISSED** [review]

**Review by LoicBrassart — DISMISSED** [review]
Much better, no more Evil !

**Review by Fbouchezpro — APPROVED** [review]

**Review by LoicBrassart — APPROVED** [review]
No More Evil

**Pull request merged**